### ISSUE TITLE
UIkit, the missing JS docs

### ISSUE THREAD

**Issue by fabien** [uikit/uikit#2629]
### What is Expected?

The documentation related to JS usage is incomplete, to the point where it's cryptic to actually use the instructions provided: https://getuikit.com/docs/javascript#programmatic-use

There's these examples:

```javascript
// Updates the component itself.
component.$emit(event = 'update');

// Updates the component itself and components nested beneath the component.
component.$update(event = 'update');

// Updates the component itself and components attached to the component's parents.
component.$update(event = 'update', parents = true);
```

It is never shown how to obtain a reference to `component`, and unless you've attached the component manually using one of the UIkit.* factory functions, it's currently not possible to obtain it later.

If you used the recommended attribute approach, but would like to call a method on a component later, you're out of luck. That is, there's a reference stored on the DOM node, as part of a private property called `__uikit__`. You _could_ grab it there, but I'd rather see a helper like this:

```javascript
$('#offcanvas').uikit('offcanvas', options); // initialize or update - returns component instance
...
var component = $('#offcanvas').uikit('offcanvas'); // grab the component for use
```

So unless the provided example `UIkit.offcanvas('#offcanvas').toggle();` works as the code above, I think such a function would be in order. For example, it's not clear if it will re-initialize or re-configure if needed, forcing an update if already attached.

That said, there's a lot of stuff going on behind the scenes with UIkit, that is fully undocumented to us. From the looks of it, it appears like an advanced UI framework, which mimics vue.js and similar projects. There's directives, attributes, local state and even automatic mounting using DOM events.

It's very hard to follow the flow and lifecycle events a UIkit component goes through, so any documentation on this would be very welcome.

In fact, it would be interesting to see a simple 'hello world' component example. Not only to learn more about the current components, but perhaps also to develop custom components using these concepts.

**Comment by florianletsch** [comment]
You are right, we should work on the docs in that direction. To get you started, here is a brief answer anyway.

> You could grab it there, but I'd rather see a helper like this: (...) So unless the provided example `UIkit.offcanvas('#offcanvas').toggle();` works as the code above

It does! Just call the component constructor on that DOM element, without providing the options argument. For example for the off-canvas:

```
var component = UIkit.offcanvas('#offcanvas-toggle');
component.toggle();
```

For the remaining requests, we will try to provide more extensive docs.

**Comment by fabien** [comment]
@florianletsch thanks for confirming this! When using this, I assume the existing component is used or updated, instead of throwing it away and re-initializing it?

UIkit is pretty awesome! Would definitely like to know more!

**Comment by badpenguin** [comment]
Does someone mind after years to answer the question? :)

**Comment by janschoenherr** [comment]
The docs have been updated, thanks. uikit/uikit-site@50d07ef0c777b752fd13b670ec9d5c50ba0daa20
